client-lib/src/main/java/com/ibm/ws/repository/connections/internal/AbstractRepositoryConnection.java

```diff
@@ -245,6 +246,33 @@ public RepositoryResource getResource(String id) throws RepositoryBackendExcepti
     }
 
     @Override
+    public Collection<RepositoryResource> getMatchingResources(FilterPredicate... predicates) throws RepositoryBackendException {
+
+        ResourceCollector<RepositoryResource> resources =
```

Comment:
I don't think you need to use a ResourceCollector here. A standard collection should be fine.

ResourceCollector (IIRC) is for doing de-duplication.

Using as you've done shouldn't hurt but I don't think you need it.